### PR TITLE
Wallet components

### LINKED ISSUE #6: Add WalletConnect

### LINKED ISSUE #40: Wallet Selection Screen
https://www.figma.com/file/meufX9Gz73ox6LCzI7eoEe/AirSwap-Low-Fidelity?node-id=53%3A36

### PR DESCRIPTION
**Pull request by gpxl-dev**
Closes #40
Closes #6

## Summary

- Adds `WalletButton` component which renders a "connect wallet" button, or a connected wallet status with a blocky identicon. Has a temporary (pending designs) disconnect "x" icon to disconnect.
- Adds `WalletProviderList` coponent to render all supported wallet providers
- Adds WalletConnect support.
- Updates `Wallet.tsx` to use both of the above, plus `react-modal` (styling temporary) to render wallet provider list within.
- Adds support for reconnecting to the previous wallet provided when the page reloads. Also stops this happening when wallet is disconnected.

## Notes

- As with all other branches currently, styling and structure is temporary
- Tests and stories are quite limited as these components are very basic.
- In order to use walletConnect, you will need to supply RPC URLs. The easiest way to do this is to create an infura account, create a project, and copy the URLs listed under "endpoints". You can then copy the contents of `.env` into your own `.env.local` file and populate the endpoints.

## Components added

- [x] WalletButton
   - [x] Connection
   - [x] Disconnection 
- [x] WalletProviderList

### CODE REVIEW

**Comment by gpxl-dev** [comment]
Will be covering #6 in this PR.

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -26,21 +31,28 @@ export type ButtonProps = {
...
-  neutral: "bg-black text-white dark:bg-white dark:text-black",
+  neutral: "bg-gray-200 text-black dark:bg-gray-700 dark:text-white",
   primary: "bg-primary text-white",
   positive: "bg-green-700 text-white",
   destructive: "bg-red-700 text-white",
 };
 
+const variantClasses: Record<ButtonVariant, string> = {
+  centered: "justify-center",
+  "left-justified": "justif",
```
Comment (hmalik88) [inline src/components/Button/Button.tsx:42]: Not sure if you were going for justify-start as left-justified's value, but justif seems to be a typo.

**Comment by codyenokida** [comment]
Text color does not reflect light/dark mode. Other than that looks good!

File: src/components/Button/Button.tsx (hunk 1 above)
Comment (gpxl-dev) [reply src/components/Button/Button.tsx:42]: Good catch, thanks!

**Comment by gpxl-dev** [comment]
@codyenokida can you clarify which text colour?

Primary button stays white on AirSwap blue in both light and dark at the moment, but the "neutral" connect wallet buttons change between white on dark grey and black on light grey.

Is it something in storybook maybe? Dark theme isn't quite set up properly on there.

**Review by hmalik88 — APPROVED** [review]
LGTM

**Review by codyenokida — APPROVED** [review]

**Pull request merged**